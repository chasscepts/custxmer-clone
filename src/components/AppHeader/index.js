import { Link, useLocation } from 'react-router-dom';
import MainMenu from '../MainMenu';
import SocialMedia from '../SocialMedia';
import css from './style.module.css';
import logo from '../../assets/images/custxmer-logo.webp';
import { formatPath } from '../../utils';

const AppHeader = () => {
  const { pathname } = useLocation();

  return (
    <header>
      <h1 className="clip">Custxmer</h1>
      <div className={css.header}>
        <Link to="/">
          <img className={css.logo} alt="logo" src={logo} />
        </Link>
        <div className={css.mainMenuWrap}>
          <MainMenu />
        </div>
        <SocialMedia />
      </div>
      <div className={css.location}>
        {formatPath(pathname)}
      </div>
    </header>
  );
};

export default AppHeader;
