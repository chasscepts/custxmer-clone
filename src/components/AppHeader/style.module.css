.header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 15px;
}

.logo {
  width: 120px;
}

.mainMenuWrap {
  display: block;
}

