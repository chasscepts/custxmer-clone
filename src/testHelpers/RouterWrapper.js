import { MemoryRouter } from "react-router";
import PropTypes from 'prop-types';
import { paths } from "../utils";

const RouterWrapper = ({ children }) => (
  <MemoryRouter initialEntries={Object.values(paths)} initialIndex={0}>
    {children}
  </MemoryRouter>
);

RouterWrapper.propTypes = {
  children: PropTypes.node,
};

RouterWrapper.defaultProps = {
  children: null,
};

export default RouterWrapper;
