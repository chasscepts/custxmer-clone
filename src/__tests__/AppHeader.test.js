import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import RouterWrapper from '../testHelpers/RouterWrapper';
import AppHeader from '../components/AppHeader';

const stubFunction = () => {};

describe('AppHeader', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(
      <RouterWrapper>
        <AppHeader setDrawerOpen={stubFunction} />
      </RouterWrapper>
    );
    expect(tree.toJSON()).toMatchSnapshot();
    tree.unmount();
  });

  describe('navigation', () => {
    beforeEach(() => {
      render(
        <RouterWrapper>
          <AppHeader setDrawerOpen={stubFunction} />
        </RouterWrapper>
      );
    });

    it('displays the path', () => {
      expect(screen.getByText('/Home')).toBeInTheDocument();
    });

    it('changes path when link is clicked', () => {
      userEvent.click(screen.getAllByRole('link', { name: 'ABOUT' })[0]);
      expect(screen.getByText('/About')).toBeInTheDocument();
    });

    it('highlights the active link', () => {
      const node = screen.getAllByRole('link', { name: 'ABOUT' })[0]
      userEvent.click(node);
      expect(node).toHaveClass('active');
    });
  });
});
