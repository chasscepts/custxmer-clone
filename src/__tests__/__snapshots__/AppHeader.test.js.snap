// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader matches snapshot 1`] = `
<header>
  <h1
    className="clip"
  >
    Custxmer
  </h1>
  <div
    className="headerWrap"
  >
    <div
      className="container header"
    >
      <div
        className="logoWrap"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <img
            alt="logo"
            className="logo"
            src="custxmer-logo.webp"
          />
        </a>
      </div>
      <div
        className="mainMenuWrap"
      >
        <nav
          className="menu"
        >
          <ul
            className="plain-list horizontal"
          >
            <li
              className="menuItem"
            >
              <a
                aria-current={true}
                className="link active"
                href="/"
                onClick={[Function]}
              >
                HOME
              </a>
            </li>
            <li
              className="menuItem"
            >
              <a
                aria-current={false}
                className="link "
                href="/about"
                onClick={[Function]}
              >
                ABOUT
              </a>
            </li>
            <li
              className="menuItem"
            >
              <a
                aria-current={false}
                className="link "
                href="/services"
                onClick={[Function]}
              >
                SERVICES
              </a>
            </li>
            <li
              className="menuItem"
            >
              <a
                aria-current={false}
                className="link "
                href="/contact"
                onClick={[Function]}
              >
                CONTACT
              </a>
            </li>
          </ul>
        </nav>
      </div>
      <div
        className="socialMediaWrap"
      >
        <nav
          className="socialMediaNav"
        >
          <ul
            className="plain-list horizontal"
          >
            <li
              className="listItem"
            >
              <a
                className="socialMediaLink"
                href="https://twitter.com/chasscepts"
                rel="noreferrer"
                target="_blank"
              >
                <i
                  className="fab fa-twitter"
                />
              </a>
            </li>
            <li
              className="listItem"
            >
              <a
                className="socialMediaLink"
                href="https://www.linkedin.com/in/chasscepts/"
                rel="noreferrer"
                target="_blank"
              >
                <i
                  className="fa-brands fa-linkedin-in"
                />
              </a>
            </li>
          </ul>
        </nav>
      </div>
      <div
        className="hamburgerWrap"
      >
        <button
          className="hamburger"
          onClick={[Function]}
          type="button"
        >
          <i
            className="fa-solid fa-bars"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    className="location"
  >
    <div
      className="container locationInner"
    >
      <span
        className="locationText"
      >
        /Home
      </span>
    </div>
  </div>
  <div
    className="drawerNavContainer "
  >
    <div
      className="drawerHeader"
    >
      <button
        className="drawerCloseBtn"
        onClick={[Function]}
        type="button"
      >
        <i
          className="fa-solid fa-xmark"
        />
      </button>
    </div>
    <nav>
      <ul
        className="plain-list"
      >
        <li
          className="listItem"
        >
          <a
            className="link active"
            href="/"
            onClick={[Function]}
          >
            HOME
          </a>
        </li>
        <li
          className="listItem"
        >
          <a
            className="link "
            href="/about"
            onClick={[Function]}
          >
            ABOUT
          </a>
        </li>
        <li
          className="listItem"
        >
          <a
            className="link "
            href="/services"
            onClick={[Function]}
          >
            SERVICES
          </a>
        </li>
        <li
          className="listItem"
        >
          <a
            className="link "
            href="/contact"
            onClick={[Function]}
          >
            CONTACT
          </a>
        </li>
      </ul>
    </nav>
  </div>
</header>
`;
